prompt_toolkit/shortcuts/progress_bar/base.py

```diff
@@ -359,6 +364,13 @@ def item_completed(self) -> None:
         self.items_completed += 1
         self.progress_bar.invalidate()
 
+    async def _remove_when_done_async(self):
+        def run_remove_when_done_thread() -> None:
+            return self.remove_when_done(self)
```

Comment:
This doesn't correspond with the type signature, I think it's fine to pass ``self``, but the ``Callable`` in ``__init__`` should match.

Reply:
hm I think I got this figured out